src/libAtomVM/nifs.c

```diff
@@ -1184,13 +1215,21 @@ static NativeHandlerResult process_console_mailbox(Context *ctx)
 
 // Common handling of spawn/1, spawn/3, spawn_opt/2, spawn_opt/4
 // opts_term is [] for spawn/1,3
-static term do_spawn(Context *ctx, Context *new_ctx, term opts_term)
+static term do_spawn(Context *ctx, Context *new_ctx, size_t arity, size_t n_freeze, term opts_term)
 {
     term min_heap_size_term = interop_proplist_get_value(opts_term, MIN_HEAP_SIZE_ATOM);
     term max_heap_size_term = interop_proplist_get_value(opts_term, MAX_HEAP_SIZE_ATOM);
     term link_term = interop_proplist_get_value(opts_term, LINK_ATOM);
     term monitor_term = interop_proplist_get_value(opts_term, MONITOR_ATOM);
     term heap_growth_strategy = interop_proplist_get_value_default(opts_term, ATOMVM_HEAP_GROWTH_ATOM, BOUNDED_FREE_ATOM);
+    term request_term = interop_proplist_get_value(opts_term, REQUEST_ATOM);
+    term group_leader;
+
+    if (UNLIKELY(request_term != term_nil())) {
+        group_leader = term_get_tuple_element(request_term, 3);
```

Comment:
if for any reason a non tuple `request` option is used as opt, this will crashes. Let's check `term_is_tuple(request_term)`.
Also let's check that `term_get_tuple_element(request_term, 3)` is a pid.